text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,405 @@
+branches. Our goal is to minimize the cost of maintaining multiple major versions at the same time, while enabling
+customers to migrate to a new major version release at their own pace.
+
+To this effect, we define the following life-cycle stages for libraries and tools (later on referenced generically as
+_software_) that are part of the CDK ecosystem:
+
+- **Developer Preview**: During this phase, software is not supported, should not be used in production environments,
+  and are meant for early access and feedback purposes only. It is possible for future releases to introduce breaking
+  changes. Once AWS identifies a release to be a stable product, it may mark it as a _Release Candidate_. _Release
+  Candidates_ are ready for _General Availability_ unless significant bugs emerge, and will receive full AWS support.
+- **Full Support**: During this phase, software is fully supported. AWS will provide regular releases that include new
+  features, support for new services, API updates for existing services, as well as bug and security fixes.
+- **Maintenance Announcement**: AWS will make a public announcement at least 6 months before software enters the
+  **Maintenance** phase. During this period, software will continue to be fully supported. Typically, this announcement
+  happens at the same time as the next major version is introduced.
+- **Maintenance**: During this phase, AWS limits releases to address critical bug fixes and security issues only.
+  Software will no longer receive API updates for new or existing services, or be updated to support new regions. The
+  **Maintenance** phase has a default duration of 12 months, unless otherwise specified as part of the **Maintenance
+  Announcement**.
+- **End-of-Support**: This phase starts at the scheduled end of the **Maintenance** phase. Software that has reached
+  **End-of-Support** will no longer receive updates or new releases. Previously published releases will continue to be
+  available via public package managers and the code will remain accessible on GitHub. The GitHub repository may be
+  archived. Use of software that has reached **End-of-Support** is done at the user's discretion. We recommend users
+  upgrade to the latest major version (the release in **Full Support** phase).
+
+Here is a visual illystration of the major version life-cycle (timelines are purely illustrative):
+
+![Illustration of the Maintenance Policy](../images/MaintenancePolicy.png)
+
+# Rationale and Alternatives
+
+## Rationale
+
+The main motivation for moving forward with a v2 release now is the desire to start moving towards the single module
+structure. The current multi module structure is known to cause a lot of confusion for users. For more details on why
+aws-cdk is moving to a single package, see the [MonoCDK RFC](https://github.com/aws/aws-cdk-rfcs/issues/6).
+
+This belief that releasing a single package will make user's lives easier is countered by a strong desire to limit the
```

Comment:
The main motivation for mono-cdk is to allow 3rd party libraries to declare CDK as a peer dependency without causing breaking changes any time additional transitive dependencies are added upstream. Basically monolithic packaging is the only viable solution.

Reply:
This paragraph isn't about "why monoCDK" (this is explained in the MonoCDK RFC as pointed to the line before), but more "why is this v2" / "why is v2 the way it is"